### PR TITLE
executor: set scanConcurrency varialbe for distsql Executor on building.

### PR DESCRIPTION
**Pull request by coocood**
Prevent worker goroutines from reading session variable map, Fixes race.

### CODE REVIEW

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -316,27 +316,5 @@ func (s *SessionVars) GetTiDBSystemVar(ctx context.Context, name string) (string
...
-	if err != nil {
-		if key == DistSQLScanConcurrencyVar {
-			// Clean up.
-			delete(s.systems, DistSQLScanConcurrencyVar)
-		}
-		return "", errors.Trace(err)
-	}
-	s.systems[key] = globalVal
-	return globalVal, nil
+	return SysVars[key].Value, nil
```
Comment (zimulala) [inline sessionctx/variable/session.go]: Does it need to consider that the 'initing' value is true？

**Comment by ngaut** [comment]
LGTM

File: sessionctx/variable/session.go (hunk 1 above)
Comment (shenli) [inline sessionctx/variable/session.go]: I have the same concern. Have you tested this branch to bootstrap a empty storage?

Comment (shenli) [inline sessionctx/variable/session.go]: And you should also update the comment of this function.

Comment (shenli) [inline sessionctx/variable/session.go]: In the bootstrapping stage, we will never access global variable table and use default value for the variable.

**Pull request merged**